src/liberty/libertyProject.ts

```diff
@@ -132,38 +133,60 @@ export class ProjectProvider implements vscode.TreeDataProvider<LibertyProject>
 				}
 			}).catch((err: any) => console.error("Unable to parse build.gradle: " + gradlePath + "; " + err));
 		}
-
 		return validGradleBuildFiles;
 	}
 
+	private projectRootPathExists(path: string, keys: Iterable<string>): Boolean {
+		for (const existingPath of keys) {
+			if ( vscodePath.dirname(existingPath) === path) {
+				return true;
+			}
+		  }
+		return false;
+	}
+
+	/**
+	 * Checks if given build file exists in existing project map (<code>this.projects</code>).
+	 * If it exists, it will be added to the given <code>projectMap</code>.
```

Comment:
I find this description of this method a little confusing. Is this to say that if the build file already exists in `this.projects`, the corresponding `LibertyProject` will be updated in the passed `projectsMap`?

I know this is old code, but rather than updating in the passed `projectsMap` value, which is really `newProjectsMap` (https://github.com/OpenLiberty/liberty-tools-vscode/pull/127/files#diff-5f236a0e97f15adc9719084cb9e2e4fe818e88043da51ff05c25d585e27eec26R183), which then overwrites `this.projects` (https://github.com/OpenLiberty/liberty-tools-vscode/pull/127/files#diff-5f236a0e97f15adc9719084cb9e2e4fe818e88043da51ff05c25d585e27eec26R241) could we instead update `this.projects` directly here?

Reply:
Will update comments.  For updating `this.project` directly, there is no real benefit of doing that, plus, it won't work, the map also holds the newly created  `LibertyProject` object.

Reply:
Thanks @travis1111, looking into this a bit more I realized why we overwrite `this.projects` with the `newProjectsMap`. 

As an FYI for anyone following this thread, at the end of the method we overwrite `this.projects` to ensure we have the most updated list of projects. That way if a project is now "outdated", ie. the LMP or LGP config has been removed from the build file we no longer display it on the dashboard.
https://github.com/OpenLiberty/liberty-tools-vscode/blob/ddeee91b4a8b73b118180109f8fc000f5c5d3544/src/liberty/libertyProject.ts#L195